crates/uv/src/commands/project/sync.rs

```diff
@@ -84,8 +86,50 @@ pub(crate) async fn sync(
         concurrency,
     );
 
-    // TODO(konsti): Read editables from lockfile.
-    let editables = ResolvedEditables::default();
+    let site_packages = SitePackages::from_executable(&venv)?;
+
+    // Build any editables.
+    let editables = {
+        let editables = resolution
+            .distributions()
+            .filter_map(|dist| {
+                if let ResolvedDist::Installable(Dist::Source(SourceDist::Directory(
+                    DirectorySourceDist {
+                        path,
+                        url,
+                        editable: true,
+                        ..
+                    },
+                ))) = dist
+                {
+                    Some(EditableRequirement {
+                        url: url.clone(),
+                        path: path.clone(),
+                        extras: vec![],
+                        marker: None,
+                        origin: None,
+                    })
+                } else {
+                    None
+                }
+            })
+            .collect::<Vec<_>>();
```

Comment:
We really need to improve the situation around editables, this is not good. But it's out of scope for this PR.